quilt3_local/quilt3_local/api.py

```diff
@@ -0,0 +1,51 @@
+import ariadne.asgi
+import boto3
+import fastapi
+from botocore.exceptions import ClientError
+
+from .graphql import schema as graphql_schema
+
+
+class ApiException(Exception):
+    """
+    Base class for API exceptions.
+    """
+    def __init__(self, status_code, message):
+        super().__init__()
+        self.status_code = status_code
+        self.message = message
+
+
+sts_client = boto3.client("sts")
+session_cred = boto3._get_default_session().get_credentials()
+
+
+api = fastapi.FastAPI()
+
+
+@api.get("/api/auth/get_credentials")
+def get_credentials():
+    """
+    Obtains credentials corresponding to your role.
+
+    Returns a JSON object with three keys:
+        AccessKeyId(string): access key ID
+        SecretAccessKey(string): secret key
+        SessionToken(string): session token
+        Expiration(ISO date string)
+    """
+    try:
+        if session_cred.token:
```

Comment:
Would be cool if we could make local catalog work (at least partially) without any AWS credentials.

If it's not doable in this PR, we should clarify in README that AWS credentials are still needed. Probably we also should handle absent credentials case more gracefully, currently it crashes with
```
    if session_cred.token:
AttributeError: 'NoneType' object has no attribute 'token'
```

Reply:
> Would be cool if we could make local catalog work (at least partially) without any AWS credentials.

not sure that would work

> Probably we also should handle absent credentials case more gracefully, currently it crashes with

done

> If it's not doable in this PR, we should clarify in README that AWS credentials are still needed

will do